### ISSUE TITLE
An example of external texture data compression

### ISSUE THREAD

**Issue by leavittx** [GPUOpen-Tools/compressonator#226]
I'm trying to compress a texture which is being initialized in run-time, instead of using `CMP_LoadTexture`:
```
                void* data = ...;
                size_t dataSize = ...;

                CMP_MipSet srcMipSet;
		memset(&srcMipSet, 0, sizeof(CMP_MipSet));
		srcMipSet.m_nWidth = avctx->width;
		srcMipSet.m_nHeight = avctx->height;
		srcMipSet.m_nDepth = 1;
		srcMipSet.m_format = CMP_FORMAT_RGBA_8888;
		srcMipSet.dwDataSize = dataSize;
		srcMipSet.pData = (CMP_BYTE*)data;

		srcMipSet.m_Flags = MS_FLAG_Default;
		srcMipSet.dwWidth = srcMipSet.m_nWidth;
		srcMipSet.dwHeight = srcMipSet.m_nHeight;
		srcMipSet.m_ChannelFormat = CF_8bit;
		srcMipSet.m_dwFourCC = 0;
		srcMipSet.m_nMipLevels = 0;
		srcMipSet.m_TextureType = TT_2D;

                CMP_MipSet dstMipSet;
		memset(&dstMipSet, 0, sizeof(CMP_MipSet));

                CMP_ERROR status = CMP_ProcessTexture(&srcMipSet, &dstMipSet, kernel_options, NULL);
                ...
```

And that returns status = -1 (ABORTED).
Can you give an example of how it's possible to compress user texture data through the Compressonator Framework API?

It will be also nice if I could provide an output data pointer like 
```
dstMipSet.pData = (CMP_BYTE*)outData;
```
so I don't need to memcpy / free afterwards.

**Comment by denislevesqueAMD** [comment]
@leavittx When creating a MipSet it is recommended to use the function `CMP_CreateMipSet` so that all of the internal structures are properly set up. Then, the current best way to use your external texture data would be to copy the pixels into the base MipLevel of the created MipSet.

We will work on adding a framework example to the code for this case, but here is a snippet to explain the recommended procedure:

    // User data
    uint32_t width = 1024;
    uint32_t height = 1024;
    uint32_t srcDataSize = 1024*1024*sizeof(uint32_t);
    uint32_t srcData = ...;
    
    // initialize MipSet
    CMP_MipSet srcMipSet = {};
    CMP_CreateMipSet(&srcMipSet, width, height, 1, CF_8bit, TT_2D);
    
    // Copy data into the first MipLevel
    CMP_MipLevel* baseLevel = 0;
    CMP_GetMipLevel(&baseLevel, &srcMipSet, 0, 0);
    
    memcpy(baseLevel->m_pbData, srcData, srcDataSize);

**Comment by leavittx** [comment]
Thanks! It would be more useful without the need of data copying though (for both input and output, so we could supply user defined output data pointer) :)

**Comment by NavNTCMP** [comment]
@leavittx  Issue addressed in v4.5 release
